src/coffee/directives/api/utils/model-key.coffee

```diff
@@ -135,4 +140,18 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
           return
         #set isInit to true to force redraw after all updates are processed
         child.updateModel model
+
+
+      modelsLength: =>
+        len = 0
+        return len unless @scope.models?
+
+        # isArray .. get length
+        # OR check to see if there is length field which (is essentially cached on the front or backend)
+        if angular.isArray(@scope.models) or @scope.models.length?
+          len = @scope.models.length
```

Comment:
@zacronos due to your advice `.length`  is emphasized to be predefined via array, backend, or frontend calculation. The worst case is an object with `length` field.